### PR TITLE
Merge ign-rendering6 ➡️  gz-rendering7

### PR DESCRIPTION
**Pull request by azeey**
# ➡️  Forward port

Port `ign-rendering6 ` ➡️  `gz-rendering7`

Branch comparison: https://github.com/ignitionrobotics/ign-rendering/compare/gz-rendering7...ign-rendering6

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: test/common_test/Mesh_TEST.cc [hunk 1]
```diff
@@ -83,6 +85,81 @@ TEST_F(MeshTest, MeshSubMesh)
   engine->DestroyScene(scene);
 }
 
+/////////////////////////////////////////////////
+TEST_F(MeshTest, MeshSkeleton)
```
Comment (azeey) [inline test/common_test/Mesh_TEST.cc:89]: This is a new test that was added in `ign-rendering6`. I had to modify it to match how we're doing common tests in `gz-rending7`. I'd appreciate more scrutiny here.

File: ogre2/src/Ogre2RenderEngine.cc [hunk 2]
```diff
@@ -415,7 +415,6 @@ void Ogre2RenderEngine::CreateLogger()
 //////////////////////////////////////////////////
 void Ogre2RenderEngine::CreateContext()
 {
-#if !defined(__APPLE__) && !defined(_WIN32)
```
Comment (azeey) [inline ogre2/src/Ogre2RenderEngine.cc:418]: This was removed in #646 back in June, but it hadn't been merged forward. Is what I did the correct thing to do?

Comment (mjcarroll) [reply ogre2/src/Ogre2RenderEngine.cc:418]: Yes, this looks correct to me :+1:

File: test/common_test/Mesh_TEST.cc (hunk 1 above)
Comment (mjcarroll) [reply test/common_test/Mesh_TEST.cc:89]: At least from CI results, seems like there are some issues.

Comment (azeey) [reply test/common_test/Mesh_TEST.cc:89]: There were a few files that I missed to commit. Should hopefully pass now.

**Pull request merged**